### PR TITLE
refactor: remove if statement in nginx.conf

### PR DESCRIPTION
**Pull request by monkeyDluffy6017**
### Description

If is evil in nginx, remove if statement in nginx.conf

### Checklist

- [x] I have explained the need for this PR and the problem it solves
- [x] I have explained the changes or the new features added to this PR
- [x] I have added tests corresponding to this change
- [ ] I have updated the documentation to reflect this change
- [x] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

File: apisix/init.lua [hunk 1]
```diff
@@ -263,6 +263,16 @@ local function set_upstream_headers(api_ctx, picked_server)
     if x_forwarded_host then
         api_ctx.var.var_x_forwarded_host = x_forwarded_host
     end
+
+    local x_forwarded_for = api_ctx.var.http_x_forwarded_for
+    if x_forwarded_for then
+        api_ctx.var.http_x_forwarded_for = x_forwarded_for .. ", " .. api_ctx.var.realip_remote_addr
```
Comment (tokers) [inline apisix/init.lua:269]: One can remove the var value in serverless plugins, so please check the value before concating it.

Comment (monkeyDluffy6017) [reply apisix/init.lua]: Done

**Comment by tzssangglass** [comment]
LGTM

**Review by tzssangglass — DISMISSED** [review]

File: t/plugin/proxy-rewrite3.t [hunk 2]
```diff
@@ -455,3 +455,99 @@ passed
 GET /echo HTTP/1.1
 --- response_headers
 test: test_in_set
+
+
+
+=== TEST 21: set route (test if X-Forwarded-Port can be set before proxy)
```
Comment (spacewander) [inline t/plugin/proxy-rewrite3.t:461]: This PR contains https://github.com/apache/apisix/pull/8404. We can deal with it after #8404 is merged.

Comment (monkeyDluffy6017) [reply t/plugin/proxy-rewrite3.t:461]: @spacewander could you check this PR again, the https://github.com/apache/apisix/pull/8404 is closed now, and all ci passed

Comment (monkeyDluffy6017) [reply t/plugin/proxy-rewrite3.t:461]: `X-Forwarded-For` is added like other `X-Forwarded-*` header

File: apisix/core/ctx.lua [hunk 3]
```diff
@@ -201,6 +201,7 @@ do
         var_x_forwarded_proto = true,
         var_x_forwarded_port  = true,
         var_x_forwarded_host  = true,
+        var_x_forwarded_for  = true,
```
Comment (spacewander) [inline apisix/core/ctx.lua:204]: Bad indent

File: apisix/init.lua [hunk 4]
```diff
@@ -267,6 +267,17 @@ local function set_upstream_headers(api_ctx, picked_server)
         api_ctx.var.var_x_forwarded_host = x_forwarded_host
     end
 
+    local x_forwarded_for = api_ctx.var.http_x_forwarded_for
```
Comment (spacewander) [inline apisix/init.lua:270]: Would be better to use http://nginx.org/en/docs/http/ngx_http_proxy_module.html#var_proxy_add_x_forwarded_for directly?

File: apisix/core/ctx.lua (hunk 3 above)
Comment (monkeyDluffy6017) [reply apisix/core/ctx.lua]: Done

File: apisix/init.lua (hunk 4 above)
Comment (monkeyDluffy6017) [reply apisix/init.lua]: Done

File: apisix/init.lua [hunk 5]
```diff
@@ -267,6 +267,11 @@ local function set_upstream_headers(api_ctx, picked_server)
         api_ctx.var.var_x_forwarded_host = x_forwarded_host
     end
 
+    local x_forwarded_for = api_ctx.var.proxy_add_x_forwarded_for
+    if x_forwarded_for then
+        api_ctx.var.var_x_forwarded_for = x_forwarded_for
```
Comment (spacewander) [inline apisix/init.lua:272]: Maybe we can set `proxy_set_header   X-Forwarded-For      $proxy_add_x_forwarded_for` directly?

Comment (monkeyDluffy6017) [reply apisix/init.lua:272]: so we don't need `var_x_forwarded_for`?

Comment (spacewander) [reply apisix/init.lua:272]: Yes. It seems there are no builtin replacements for the others.

Comment (monkeyDluffy6017) [reply apisix/init.lua]: Done

**Review by leslie-tsang — APPROVED** [review]

**Review by soulbird — APPROVED** [review]

**Pull request merged**